billing/customer/service_test.go

```diff
@@ -0,0 +1,362 @@
+package customer_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/raystack/frontier/billing"
+	"github.com/raystack/frontier/billing/customer"
+	"github.com/raystack/frontier/billing/customer/mocks"
+	stripemock "github.com/raystack/frontier/billing/stripetest/mocks"
+	"github.com/stripe/stripe-go/v79"
+	"github.com/stripe/stripe-go/v79/client"
+)
+
+func mockService(t *testing.T) (*client.API, *stripemock.Backend, *mocks.Repository) {
+	t.Helper()
+	mockRepository := mocks.NewRepository(t)
+	mockBackend := stripemock.NewBackend(t)
+
+	stripeClient := client.New("key_123", &stripe.Backends{
+		API: mockBackend,
+	})
+
+	return stripeClient, mockBackend, mockRepository
+}
+
+func TestService_Create(t *testing.T) {
+	ctx := context.Background()
+	type args struct {
+		customer customer.Customer
+		offline  bool
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    customer.Customer
+		wantErr bool
```

Comment:
Can we have a field which also indicates what the error is? Currently, it is not very clear what error we would get in different scenarios.
For example, in the first one below, the list API does not return any error. The params sent are valid (barring the duplicacy in orgID). However, we still get an error. It would be better if we could even specify what error to expect in that situation.

Same for all other tests where we have not specified the error type.

Reply:
Here's why this is important:
Unit tests can often act as a documentation of sorts or help in debugging even, if someone completely new to the system tries to look into the code.

In our services even if we simply return back the error raised by repositories, even then, it is essential to mock a particular error and test that that exact error is being returned. This way, whenever anyone wants to debug such an error, they would know that they need to look in the repository.